spec/lang/step/terminators.md

```diff
@@ -350,6 +353,51 @@ impl<M: Memory> Machine<M> {
 Note that the caller has no guarantee at all about the value that it finds in its return place.
 It should probably do a `Validate` as the next step to encode that it would be UB for the callee to return an invalid value.
 
+## Start Unwind
+
+```rust
+impl<M: Memory> Machine<M> {
+    fn eval_terminator(&mut self, Terminator::StartUnwind(block_name): Terminator) -> NdResult {
+        self.jump_to_block(block_name)?;
+        ret(())
+    }
+}
+```
+
+## Resume Unwind
+
+```rust
+impl<M: Memory> Machine<M> {
+    fn eval_terminator(&mut self, Terminator::ResumeUnwind: Terminator) -> NdResult {
+        let mut frame = self.mutate_cur_stack(
+            |stack| stack.pop().unwrap()
+        );
+
+        // Deallocate everything.
+        while let Some(local) = frame.locals.keys().next() {
+            frame.storage_dead(&mut self.mem, local)?;
+        }
+
+        // Inform the memory model that this call has ended.
+        self.mem.end_call(frame.extra)?;
+
+        // Terminate current thread, if the bottom of the stack is reached.
+        if self.active_thread().stack.len() == 0{
+                self.terminate_active_thread("resume")?;
+        }
```

Comment:
Having to check the stack lengths here is suspect; this should look more like the logic in `Return`.